eclair-core/src/main/scala/fr/acinq/eclair/payment/send/MultiPartPaymentLifecycle.scala

```diff
@@ -264,7 +264,8 @@ class MultiPartPaymentLifecycle(nodeParams: NodeParams, cfg: SendPaymentConfig,
           }
           paymentSent.feesPaid + localFees
       }
-      context.system.eventStream.publish(PathFindingExperimentMetrics(cfg.paymentHash, cfg.recipientAmount, fees, status, duration, now, isMultiPart = true, request.routeParams.experimentName, cfg.recipientNodeId, cfg.invoice.map(_.routingInfo)))
+      val hints_opt = cfg.invoice.flatMap(invoice => if (invoice.routingInfo.isEmpty) None else Some(invoice.routingInfo))
+      context.system.eventStream.publish(PathFindingExperimentMetrics(cfg.paymentHash, cfg.recipientAmount, fees, status, duration, now, isMultiPart = true, request.routeParams.experimentName, cfg.recipientNodeId, hints_opt))
```

Comment:
So that we put a `null` in the db in case there is no routing hint instead of an empty list.

Reply:
Right, `Bolt11Invoice.routingInfo` is a `Seq[Seq[]]`, not an `Option[Seq[Seq]]`. Which is correct because `r` is a repeated field in the invoice.

So there is an inconsistency in the model between `Bolt11Invoice.routingInfo` and `PathFindingExperimentMetrics.routingHints_opt`, but it's not a big deal and it makes the migration seamless.